### ISSUE TITLE
rke2 version v1.21.4+rke2r2 won't complete startup with the same configs as rke2 version v1.21.2+rke2r1.

### ISSUE THREAD

**Issue by MonkeyMichael** [rancher/rke2#1773]
**Environmental Info:**
RKE2 Version: 
```
rke2 version v1.21.4+rke2r2 (d5746a33b001dcf1a24cd3256e9f1df2a43ff3dd)
go version go1.16.6b7
```
Node(s) CPU architecture, OS, and Version: 
OS - Modified 18.04.5 LTS (Bionic Beaver)
```
Linux <hostname> 5.4.98-5.4.4-amd64-14794deb7c18c4e7 #1 SMP Mon Mar 22 19:54:29 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

Cluster Configuration:
This is just setting up the first node in a cluster of 3 servers and 1 agent.

**Describe the bug:**
I'm trying to use "rke2 version v1.21.4+rke2r2", and i'm having a hard time getting my first node ready with the same configuration as when I'm on "rke2 version v1.21.2+rke2r1". 
The logs are showing errors trying to connect to various ports. This seems normal at first, but with this newest version of RKE2, it never recovers.
I've tried uninstalling the node clean and reinstalling to no avail. If I downgrade the RKE2 binary to the old version, it will come up fine. 
 
 These are the errors that keep scrolling in the logs:
```
 rke2[10194]: {"level":"warn","ts":"2021-09-08T18:58:49.156Z","caller":"clientv3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"passthrough:///https://127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:2379: connect: connection refused\""}
 rke2[10194]: time="2021-09-08T18:58:49Z" level=error msg="Failed to check local etcd status for learner management: context deadline exceeded"
 rke2[10194]: {"level":"warn","ts":"2021-09-08T18:58:50.158Z","caller":"clientv3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"passthrough:///https://127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:2379: connect: connection refused\""}
 rke2[10194]: time="2021-09-08T18:58:50Z" level=info msg="Failed to test data store connection: context deadline exceeded"
 rke2[10194]: time="2021-09-08T18:58:50Z" level=info msg="Cluster-Http-Server 2021/09/08 18:58:50 http: TLS handshake error from 127.0.0.1:51685: remote error: tls: bad certificate"
 rke2[10194]: time="2021-09-08T18:58:50Z" level=info msg="Waiting to retrieve kube-proxy configuration; server is not ready: https://127.0.0.1:9345/v1-rke2/readyz: 500 Internal Server Error"
 rke2[10194]: time="2021-09-08T18:58:51Z" level=warning msg="Unable to watch for tunnel endpoints: Get \"https://127.0.0.1:6443/api/v1/namespaces/default/endpoints?fieldSelector=metadata.name%3Dkubernetes&resourceVersion=0&watch=true\": dial tcp 127.0.0.1:6443: connect: connection refused"
 rke2[10194]: time="2021-09-08T18:58:54Z" level=info msg="Waiting for API server to become available"
 rke2[10194]: time="2021-09-08T18:58:54Z" level=info msg="Waiting for etcd server to become available"
 ```
 This is what the config.yaml file looks like for this first node.
 ```
 disable:
  - "rke2-canal"
  - "rke2-kube-proxy"
data-dir: "/kube/rancher/rke2"
kubelet-arg:
  - "root-dir=/kube/rancher/rke2/kubelet"
  - "make-iptables-util-chains=false"
  - "client-ca-file=/kube/rancher/rke2/server/tls/combined-ca.cert"
  ```
I've also tried to disable kube-proxy in a different ways by adding this:
```
 disable-kube-proxy: true
 disable:
  - "rke2-canal"
  - "rke2-kube-proxy"
  - "kube-proxy"
  ```

**Steps To Reproduce:**
For me it's just start RKE2 with version v1.21.4+rke2r2.

**Expected behavior:**
First RKE2 node to come up as it normally should.

**Actual behavior:**
RKE2 starts up, but is unable to complete.

**Additional context / logs:**
I am using Calico eBPF with kube-proxy disabled. I have setup a kubernetes services endpoint:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: kubernetes-services-endpoint
  namespace: kube-system
data:
  KUBERNETES_SERVICE_HOST: 127.0.0.1
  KUBERNETES_SERVICE_PORT: "6443"
  ```
  I see in the logs that it's trying to "Waiting to retrieve kube-proxy configuration". Not sure if that's the problem or not.

**Comment by brandond** [comment]
It looks like for some reason etcd isn't starting:
```
 rke2[10194]: time="2021-09-08T18:58:54Z" level=info msg="Waiting for etcd server to become available"
```

Can you check the containerd logs to see why?

**Comment by MonkeyMichael** [comment]
> It looks like for some reason the etcd static pod isn't starting; that needs to happen before anything else (apiserver, kube-proxy, etc) will start.
> 
> ```
>  rke2[10194]: time="2021-09-08T18:58:54Z" level=info msg="Waiting for etcd server to become available"
> ```
> 
> Can you check the containerd logs to see why?

I assume you mean in /var/log/containers? If there's another place then please let me know. That dir only has kube-system_kube-proxy logs. Should it even be trying to start kube-proxy if I'm disabling it?

I'm using nerdctl to see the containers, and I noticed only 3 show up with the pause command
```
# nerdctl ps -a --no-trunc 
CONTAINER ID                                                        IMAGE                                     COMMAND     CREATED           STATUS    PORTS    NAMES
0c5d6f53e75c016bafd6b43fa1b821c4785bff0706a97498e2e26cee1c73e29d    private.repo.com/rancher/pause:3.5    "/pause"    14 minutes ago    Up                 
7b2c7f3f16967779a0dd735cd2455c23dfb9e4ddb18d4a91783558f0f8853fdb    private.repo.com/rancher/pause:3.5    "/pause"    14 minutes ago    Up                 
f6dfb6ad5f099eeb5efc5648298b75a11072f99c942eae08cfc1c820653396c0    private.repo.com/rancher/pause:3.5    "/pause"    14 minutes ago    Up                 
```
Also, If I update the rke2 binary to use v1.21.2+rke2r1 instead then everything comes up fine.

**Comment by brandond** [comment]
No, like literally look at the containerd log file at `/var/lib/rancher/rke2/agent/containerd/containerd.log`

**Comment by MonkeyMichael** [comment]
Ah right!

I am going to guess this might have something to do with it? Do you know what path it is looking for exactly?
```
time="2021-09-08T23:28:38.972858559Z" level=info msg="CreateContainer within sandbox \"f6dfb6ad5f099eeb5efc5648298b75a11072f99c942eae08cfc1c820653396c0\" for container &ContainerMetadata{Name:etcd,Attempt:0,}"
time="2021-09-08T23:28:39.101768041Z" level=error msg="CreateContainer within sandbox \"f6dfb6ad5f099eeb5efc5648298b75a11072f99c942eae08cfc1c820653396c0\" for &ContainerMetadata{Name:etcd,Attempt:0,} failed" error="failed to create containerd container: get apparmor_parser version: exec: \"apparmor_parser\": executable file not found in $PATH"
time="2021-09-08T23:28:44.970984850Z" level=info msg="CreateContainer within sandbox \"7b2c7f3f16967779a0dd735cd2455c23dfb9e4ddb18d4a91783558f0f8853fdb\" for container &ContainerMetadata{Name:kube-apiserver,Attempt:0,}"
time="2021-09-08T23:28:45.067831543Z" level=error msg="CreateContainer within sandbox \"7b2c7f3f16967779a0dd735cd2455c23dfb9e4ddb18d4a91783558f0f8853fdb\" for &ContainerMetadata{Name:kube-apiserver,Attempt:0,} failed" error="failed to create containerd container: get apparmor_parser version: exec: \"apparmor_parser\": executable file not found in $PATH"
time="2021-09-08T23:28:50.970611492Z" level=info msg="CreateContainer within sandbox \"f6dfb6ad5f099eeb5efc5648298b75a11072f99c942eae08cfc1c820653396c0\" for container &ContainerMetadata{Name:etcd,Attempt:0,}"
time="2021-09-08T23:28:51.066548629Z" level=error msg="CreateContainer within sandbox \"f6dfb6ad5f099eeb5efc5648298b75a11072f99c942eae08cfc1c820653396c0\" for &ContainerMetadata{Name:etcd,Attempt:0,} failed" error="failed to create containerd container: get apparmor_parser version: exec: \"apparmor_parser\": executable file not found in $PATH"
time="2021-09-08T23:28:52.969767261Z" level=info msg="CreateContainer within sandbox \"0c5d6f53e75c016bafd6b43fa1b821c4785bff0706a97498e2e26cee1c73e29d\" for container &ContainerMetadata{Name:cloud-controller-manager,Attempt:0,}"
time="2021-09-08T23:28:53.078070692Z" level=error msg="CreateContainer within sandbox \"0c5d6f53e75c016bafd6b43fa1b821c4785bff0706a97498e2e26cee1c73e29d\" for &ContainerMetadata{Name:cloud-controller-manager,Attempt:0,} failed" error="failed to create containerd container: get apparmor_parser version: exec: \"apparmor_parser\": executable file not found in $PATH"
```

**Comment by brandond** [comment]
OK - that's what I was expecting. This is a duplicate of https://github.com/rancher/rke2/issues/1459#issuecomment-886993612

**Comment by MonkeyMichael** [comment]
Yep, I was just looking at similar issues with that same solution. I'll try that and see if that works. Thanks!

**Comment by MonkeyMichael** [comment]
I installed and started up apparmor and that worked.
